src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -212,26 +215,143 @@ define(function (require, exports, module) {
     }
 
     // Send Analytics data to Server
-    function sendAnalyticsDataToServer(eventParams) {
+    function sendOrSaveAnalyticsData(eventParams) {
+        var result = new $.Deferred();
+        var unsentEventFileLocation = FileSystem.getFileForPath(brackets.app.getApplicationSupportDirectory() + "/unsentEventFile.txt");
+
+        if(window.navigator.onLine){
+            internetAvailableSendAllEvents(unsentEventFileLocation, eventParams).done(function() {
+                result.resolve();
+            });
+        }else {
+            internetUnavailableSaveEventsToDisk(unsentEventFileLocation, eventParams).done(function() {
+                result.resolve();
+            }).fail(function () {
+                result.reject();
+            });
+        }
+        return result.promise();
+    }
+
+    /*
+     * Called when internet is available. Check whether the local log file has any content
+     * If it is has, add those to event array while sending to ingest as a single request
+    */
+
+    function internetAvailableSendAllEvents(unsentEventFileLocation, eventParams) {
+        var result = new $.Deferred();
+        var analyticsData = [];
+
+        unsentEventFileLocation.exists(function (err, exists) {
+            if (err) {
+                // logging the error but will still send the current eventParams which needs to be logged
+                console.error("Error while checking if the error log file exits or not");
+                sendAnalyticsDataToServer(unsentEventFileLocation, analyticsData, eventParams).done(function () {
+                    result.resolve();
+                }).fail(function (err) {
+                    console.error("Unable to send file to server");
```

Comment:
in this scenario file is not being sent. please correct the log. also, result.reject() should be added.

Reply:
result.reject() ?